syntax = "proto3";

package protocols.navigation;

import "google/protobuf/timestamp.proto";
import "protocols/common/geometry.proto";
import "protocols/common/robot_pose.proto";
import "protocols/navigation/motion.proto";

message Navigation {
	uint32 id = 1;
	google.protobuf.Timestamp timestamp = 2;
	double forward_velocity = 3;
	double left_velocity = 4;
	double angular_velocity = 5;
	
	uint32 custom_command = 6;
	
	uint32 sequence_number = 7;
	common.Point2Df output_global_linear_velocity = 8;
	
	// Real robot only, odometry control
	oneof motion {
		GoToPoint go_to_point = 9;
		RotateInPoint rotate_in_point = 10;
		RotateOnSelf rotate_on_self = 11;
	}
	common.RobotPose robot_pose = 12;
}
